bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.java

```diff
@@ -219,45 +191,21 @@ String getApiKeyFromFilename(File file) {
 
     /**
      * Generates a filename for the Event in the format
-     * "[timestamp]_[apiKey]_[UUID][startupcrash|not-jvm].json"
+     * "[timestamp]_[apiKey]_[errorTypes]_[UUID]_[startupcrash|not-jvm].json"
      */
     @NonNull
     @Override
     String getFilename(Object object) {
-        String uuid = UUID.randomUUID().toString();
-        long now = System.currentTimeMillis();
-        return getFilename(object, uuid, null, now, storeDirectory);
-    }
-
-    String getFilename(Object object, String uuid, String apiKey,
-                       long timestamp, String storeDirectory) {
-        String suffix = "";
-
-        if (object instanceof Event) {
-            Event event = (Event) object;
-
-            Number duration = event.getApp().getDuration();
-            if (duration != null && isStartupCrash(duration.longValue())) {
-                suffix = STARTUP_CRASH;
-            }
-            apiKey = event.getApiKey();
-        } else { // generating a filename for an NDK event
-            suffix = "not-jvm";
-            if (apiKey.isEmpty()) {
-                apiKey = config.getApiKey();
-            }
-        }
-        return String.format(Locale.US, "%s%d_%s_%s%s.json",
-                storeDirectory, timestamp, apiKey, uuid, suffix);
+        EventFilenameInfo eventInfo
+                = EventFilenameInfo.Companion.fromEvent(object, null, config);
+        String encodedInfo = eventInfo.encode();
+        return String.format(Locale.US, "%s%s.json", storeDirectory, encodedInfo);
```

Comment:
The `.json` extension is currently outside the part controlled by `EventFilenameInfo` but is used internally by that class as part of the `findApiKeyInFilename` test. Suggest the extension becomes part of the suffix field so it's all controlled by `EventFilenameInfo`.

Reply:
Updated so that the extension is set in `EventFilenameInfo`